codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/error/OperationErrorGenerator.kt

```diff
@@ -28,61 +27,49 @@ import software.amazon.smithy.rust.codegen.core.rustlang.writable
 import software.amazon.smithy.rust.codegen.core.smithy.RuntimeType
 import software.amazon.smithy.rust.codegen.core.smithy.RustSymbolProvider
 import software.amazon.smithy.rust.codegen.core.smithy.customize.Section
+import software.amazon.smithy.rust.codegen.core.smithy.transformers.eventStreamErrors
+import software.amazon.smithy.rust.codegen.core.smithy.transformers.operationErrors
+import software.amazon.smithy.rust.codegen.core.util.UNREACHABLE
 import software.amazon.smithy.rust.codegen.core.util.hasTrait
 import software.amazon.smithy.rust.codegen.core.util.toSnakeCase
 
-/**
- * For a given Operation ([this]), return the symbol referring to the operation error. This can be used
- * if you, e.g. want to return an operation error from a function:
- *
- * ```kotlin
- * rustWriter.rustBlock("fn get_error() -> #T", operation.errorSymbol(symbolProvider)) {
- *     write("todo!() // function body")
- * }
- * ```
- */
-fun OperationShape.errorSymbol(symbolProvider: RustSymbolProvider): RuntimeType {
-    val operationSymbol = symbolProvider.toSymbol(this)
-    return RustModule.Error.toType().resolve("${operationSymbol.name}Error")
-}
-
-fun UnionShape.eventStreamErrorSymbol(symbolProvider: RustSymbolProvider): RuntimeType {
-    val unionSymbol = symbolProvider.toSymbol(this)
-    return RustModule.Error.toType().resolve("${unionSymbol.name}Error")
-}
-
 /**
  * Generates a unified error enum for [operation]. [ErrorGenerator] handles generating the individual variants,
  * but we must still combine those variants into an enum covering all possible errors for a given operation.
+ *
+ * This generator also generates errors for event streams.
```

Comment:
It feels weird when a functions is called out as handling two things. My thoughts are of https://en.wikipedia.org/wiki/Zero_one_infinity_rule

I don't see any obvious improvements though and I think calling out that it supports something is more helpful than not doing so.

Reply:
Conceptually, I feel like we're treating operations and event streams like two separate things but I imagine them more as:
- Operations represent some sort of request/response thing
- Operations support communication by many different protocols
- Each protocol may actually send many requests and receive many responses when run
- We hide that as an implementation detail and represent things as a single "operation request"
- event streams are one such protocol

Put another way, why call out event streams here but not the other protocols we support?

Reply:
This sentiment was addressed in [another comment](https://github.com/awslabs/smithy-rs/pull/2336#discussion_r1103247328).